### ISSUE TITLE
Add support for ECR image scanning

### ISSUE THREAD

**Issue by tomelliff** [terraform-providers/terraform-provider-aws#10656]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

[ECR repos now support image scanning](https://aws.amazon.com/blogs/containers/amazon-ecr-native-container-image-scanning/). This support is in the [v1.25.20 SDK](https://github.com/aws/aws-sdk-go/pull/2916).

### New or Affected Resource(s)

* aws_ecr_repository resource and data source

### Potential Terraform Configuration

```hcl
resource "aws_ecr_repository" "foo" {
  name = "bar"

  scan_on_push = true
}
```

### References

https://aws.amazon.com/about-aws/whats-new/2019/10/announcing-image-scanning-for-amazon-ecr/

**Comment by ewbankkit** [comment]
Requires [AWS SDK v1.25.20](https://github.com/aws/aws-sdk-go/releases/tag/v1.25.20):
* https://github.com/terraform-providers/terraform-provider-aws/pull/10476

**Comment by sergei-ivanov** [comment]
I already have a working implementation for that on my branch. Only need to update the tests and the docs, and a PR will be ready, as soon as the latest AWS SDK is merged into master.

**Comment by bflad** [comment]
Support for the below has been merged and will release with version 2.34.0 of the Terraform AWS Provider, tomorrow. Thanks to @sergei-ivanov for the quick implementation 👍 

```hcl
resource "aws_ecr_repository" "example" {
  # ... other configuration

  image_scanning_configuration {
    scan_on_push = true # or false :)
  }
}
```

**Comment by bflad** [comment]
This has been released in [version 2.34.0 of the Terraform AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#2340-october-31-2019). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
